package in.strikes.springDataJPADemo.service;

import in.strikes.springDataJPADemo.entity.Student;

import in.strikes.springDataJPADemo.repository.StudentRepository;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Sort;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class StudentService {

    StudentRepository studentRepository;
    public StudentService(StudentRepository studentRepository){
        this.studentRepository = studentRepository;
    }
  public void createStudent(Student student){

        studentRepository.save(student);
  }

  public Student fetchStudentById(Long id){
       Optional<Student> studentOptional =  studentRepository.findById(id);
       return studentOptional.get();
  }

  public List<Student> fetchAll(String name){
        Sort sort = Sort.by("age").descending();

        //pageable me pageRequest.of(starting page number, how many values in one page)
      Pageable pageable = PageRequest.of(1,2);
        Page<Student> studentList = studentRepository.findAll(pageable);
        for(Student s: studentList.getContent()){
            System.out.println(s);
        }
        return studentList.getContent();
  }


}
